jina/executors/decorators.py

```diff
@@ -381,3 +396,54 @@ def arg_wrapper(*args, **kwargs):
         return _batching(func)
     else:
         return _batching
+
+
+def single(func: Callable[[Any], np.ndarray] = None,
+           split_over_axis: int = 0,
+           merge_over_axis: int = 0,
+           slice_on: int = 1) -> Any:
+    """
+    Guarantee that the input of a function is provided as a single instance and not in batches
+
+    :param func: function to decorate
+    :param split_over_axis: split over which axis into single instances
+    :param merge_over_axis: merge over which axis into a single result
+    :param slice_on: the location of the data. When using inside a class,
+            ``slice_on`` should take ``self`` into consideration.
+    :return: the merged result as if run :func:`func` once on the input.
+
+    Example:
+        .. highlight:: python
+        .. code-block:: python
+
+            class OneByOneCrafter:
+                @single
+                def craft(self, text: str) -> Dict:
+    """
+
+    def _single(func):
+        @wraps(func)
+        def arg_wrapper(*args, **kwargs):
+            # priority: decorator > class_attribute
+            # by default data is in args[1] (self needs to be taken into account)
+            data = args[slice_on]
+            args = list(args)
+
+            default_logger.debug(
+                f'batching disabled for {func.__qualname__} axis={split_over_axis}')
+
+            final_result = []
+            for instance in data:
+                args[slice_on] = instance
+                r = func(*args, **kwargs)
+                if r is not None:
+                    final_result.append(r)
+
+            return _merge_results_after_batching(final_result, merge_over_axis)
+
+        return arg_wrapper
+
+    if func:
+        return _single(func)
+    else:
+        return _single
```

Comment:
Just curious is it possible/or under which situation to reach this line? For example, not decorating a function?